assets/js/util/i18n.js

```diff
@@ -33,28 +33,51 @@ import { __, sprintf } from '@wordpress/i18n';
  * For example, passing 65 returns '1m 5s'.
  *
  * @since 1.0.0
+ * @since n.e.x.t Refactored and renamed to improve localization.
+ * @private
  *
- * @param {number} seconds The number of seconds.
- * @return {string} Human readable string indicating time elapsed.
- *
+ * @param {number}                     seconds   The number of seconds.
+ * @param {(Intl.NumberFormatOptions)} [options] Optional formatting options.
+ * @return {string}   		Human readable string indicating time elapsed.
  */
-export const prepareSecondsForDisplay = ( seconds ) => {
+const durationFormat = ( seconds, options = { unitDisplay: 'short' } ) => {
 	seconds = parseInt( seconds, 10 );
 
-	if ( isNaN( seconds ) || 0 === seconds ) {
-		return '0.0s';
+	if ( Number.isNaN( seconds ) ) {
+		seconds = 0;
 	}
-	const results = {};
-	results.hours = Math.floor( seconds / 60 / 60 );
-	results.minutes = Math.floor( ( seconds / 60 ) % 60 );
-	results.seconds = Math.floor( seconds % 60 );
 
-	const returnString =
-		( results.hours ? results.hours + 'h ' : '' ) +
-		( results.minutes ? results.minutes + 'm ' : '' ) +
-		( results.seconds ? results.seconds + 's ' : '' );
+	let hours = Math.floor( seconds / 60 / 60 ) || '';
+	let minutes = Math.floor( ( seconds / 60 ) % 60 ) || '';
+
+	seconds = Math.floor( seconds % 60 );
+
+	if ( hours ) {
+		hours = numberFormat( hours, {
+			style: 'unit',
+			unit: 'hour',
+			...options,
+		} );
+	}
 
-	return returnString.trim();
+	if ( minutes ) {
+		minutes = numberFormat( minutes, {
+			style: 'unit',
+			unit: 'minute',
+			...options,
+		} );
+	}
+
+	seconds = numberFormat( seconds, {
+		style: 'unit',
+		unit: 'second',
+		...options,
+	} );
+
+	/* translators: 1: hours, 2: minutes, 3: seconds */
+	const formattedString = sprintf( _x( '%1$s %2$s %3$s', 'duration of time: hh mm ss', 'google-site-kit' ), hours, minutes, seconds );
```

Comment:
Let's split this onto a few lines as `sprintf` gets a bit hard to follow with the translated string and multiple parameters all on one line. This should do it (although it gets confused about the indentation of the translators comment sometimes 😄 )
```suggestion
	const formattedString = sprintf(
		/* translators: 1: hours, 2: minutes, 3: seconds */
		_x( '%1$s %2$s %3$s', 'duration of time: hh mm ss', 'google-site-kit' ),
		hours,
		minutes,
		seconds
	);
```

One additional thought here – perhaps we should document the placeholders a bit more descriptively since each one will also have its localized units included (if non-zero), so we should make it clear to avoid translators adding unit prefix/suffixes.

Reply:
@aaemnnosttv Something like `duration of time (numeric): hh mm ss` perhaps, or does it need to be more verbose than that?